frontend-web/webclient/app/Applications/Studio/Group.tsx

```diff
@@ -0,0 +1,377 @@
+import * as UCloud from "@/UCloud";
+import MainContainer from "@/MainContainer/MainContainer";
+import {Box, Button, Checkbox, DataList, Flex, Icon, Input, Label, Relative, TextArea} from "@/ui-components";
+import React, {useEffect, useState} from "react";
+import {RetrieveGroupResponse, clearLogo, setGroup, retrieveGroup, updateGroup, uploadLogo} from "../api";
+import {useCloudAPI, useCloudCommand} from "@/Authentication/DataHook";
+import {useRefreshFunction} from "@/Navigation/Redux/HeaderActions";
+import * as Heading from "@/ui-components/Heading";
+import {useNavigate, useParams} from "react-router";
+import {ButtonClass} from "@/ui-components/Button";
+import {HiddenInputField} from "@/ui-components/Input";
+import {snackbarStore} from "@/Snackbar/SnackbarStore";
+import {dialogStore} from "@/Dialog/DialogStore";
+import {AppToolLogo, groupLogoCache} from "../AppToolLogo";
+import {stopPropagation} from "@/UtilityFunctions";
+import {Tag} from "../Card";
+import {compute} from "@/UCloud";
+import ReactModal from "react-modal";
+import {largeModalStyle} from "@/Utilities/ModalUtilities";
+import {pageV2Of} from "@/DefaultObjects";
+import List, {ListRow} from "@/ui-components/List";
+
+export const AppGroup: React.FunctionComponent = () => {
+    const id = useParams<{id: string}>().id!;
+    const [group, fetchGroup] = useCloudAPI<RetrieveGroupResponse|undefined>(
+        {noop: true},
+        undefined
+    );
+
+    const [appList, setAppList] = useCloudAPI<Page<compute.ApplicationSummaryWithFavorite>|undefined>(
+        {noop: true},
+        undefined
+    );
+
+    const [, setLogoCacheBust] = useState("" + Date.now());
+
+    const groupTitleField = React.useRef<HTMLInputElement>(null);
+    const groupDescriptionField = React.useRef<HTMLInputElement>(null);
+    const appSearchField = React.useRef<HTMLInputElement>(null);
+
+    const [defaultApplication, setDefaultApplication] = useState<{name: string; version: string;}|undefined>(undefined);
+    const [tags, setTags] = useState<string[]>([]);
+    const [selectedTag, setSelectedTag] = useState<string>("");
+    const [allTags, fetchAllTags] = useCloudAPI<string[]>(
+        {noop: true},
+        []
+    );
+    const [addApplicationOpen, setAddApplicationOpen] = useState<boolean>(false);
+
+    const [commandLoading, invokeCommand] = useCloudCommand();
+
+    useEffect(() => {
+        fetchGroup(retrieveGroup({id: id}));
+        fetchAllTags(UCloud.compute.apps.listTags({}));
+        setAppList(compute.apps.listAll({}));
+    }, [id]);
+
+    const refresh = () => {
```

Comment:
I think this should be memoized, otherwise `useRefreshFunction`  will update on every potential rerender